### DISCUSSION TITLE
Get balance from seed phrase

### DISCUSSION THREAD

**Discussion by maxwellss** [1200wd/bitcoinlib#326] [Q&A] [answered]
As a beginner, this looks like a promising package.
This is more of an appeal for help, or a nudge in the right direction, than an issue.
I wondered if this package includes, or contains, code which can use a wallet's seed phrase ( 24-word ) to read its balance?
many thanks in advance,

**Comment by mccwdev** [comment, marked as answer]
Thank you.
Yes, you can import an seed phrase, create a wallet and then scan it's keys:
```
passphrase = 'never stamp rival shift ignore wrap chuckle column work price verify letter'
w = Wallet.create("Wallet", keys=passphrase, network='bitcoin')
w.scan()
w.info()
```

**Reply by maxwellss** [reply]
Hi, and many thanks for the reply.
I tested this by giving the seed phrase of an EXISTING wallet. The script worked, in the sense that it ran and printed a result.
But, the reported balances were all 0. I'm certain that the seed phrase was correct.
Perhaps I'm misunderstanding what's happening? I'm unsure whether the script is CREATING addresses and keys, or REPORTING them ?
IF it's reporting addresses and keys for an existing wallet, what are the possible reasons that balances are mis-reported ? ( or different to what I know to be the balance )
many thanks in advance

**Reply by mccwdev** [reply]
Not all wallets use the same key derivation scheme. Bitcoinlib uses the same scheme as Electrum.
You can also use different witness_types (legacy, segwit or p2sh_segwit) which results in different address types.

You can check if the generated addresses are the same. And check the derivation paths.
For instance the derivation path for the key in this wallet is m/44'/0'/0'/0/0

```
  - NETWORK: bitcoin -
  - - Keys
     40 m/44'/0'/0'/0/0              1AtmWTria8Yt2zxWHuMxHCcvEAntj38ZzL            address index 0                        0.00000000 ₿

```
